scripts/trackCutoffData.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * @fileoverview Main tracker of all cutoff data for internal storage in case Sekai.Best goes down
+ * @author Ai0796
+ */
+
+const { CUTOFF_INTERVAL, CUTOFF_DATA} = require('../constants');
+const fs = require('fs');
+
+//Cutoffs we store
+const cutoffs = [
+    1,
+    2,
+    3,
+    10,
+    20,
+    30,
+    40,
+    50,
+    100,
+    200,
+    300,
+    400,
+    500,
+    1000,
+    2000,
+    3000,
+    4000,
+    5000,
+    10000,
+    20000,
+    30000,
+    40000,
+    50000,
+];
+
+/**
+ * Recurvsively writes cutoff data to a JSON
+ * @param {Object} response from project sekai client
+*/
+async function logResults(response) {
+    try{
+        let event = getRankingEvent().id;
+        if (response['rankings'][0] != null && event != -1) {
+            let score = response['rankings'][0]['score'];
+            let rank = response['rankings'][0]['rank'];
+            let timestamp = new Date().toISOString();
+            var cutoffFile = null;
+            if (!fs.existsSync(`${CUTOFF_DATA}/Event${event}/${rank}.json`)) {
+                fs.mkdirSync(`${CUTOFF_DATA}/Event${event}`, { recursive: true });
+                cutoffFile = new Array();
+            }
+            else {
+                cutoffFile = JSON.parse(fs.readFileSync(`${CUTOFF_DATA}/Event${event}/${rank}.json`, 'utf8'));
+            }
+
+            cutoffFile.push({ 'timestamp': timestamp, 'score': score });
+
+            fs.writeFile(`${CUTOFF_DATA}/Event${event}/${rank}.json`, JSON.stringify(cutoffFile), err => {
```

Comment:
Hi Ai0, JSON is not an ideal format for processing / storing a large amount of data points.

Although we're only storing around 16560 data points a day, we should probably adhere to best practices.
Additionally, all 16k (multiplied by days into event) points are going to read from disk and loaded into memory every time this function is called which is going to be really inefficient.

Maybe look into saving this data in the SQL DB instead of json.

[Client.js](https://github.com/potor10/RoboNene/blob/master/client/client.js) has an example demonstrating how to access / use a sqlite DB
```
  loadDb(dir = CLIENT_CONSTANTS.DB_DIR) {
    this.db = new Database(`${dir}/${CLIENT_CONSTANTS.DB_NAME}`);

    // Read an encrypted database
    this.db.pragma(`key='${secretKey}'`);

    this.db.prepare('CREATE TABLE IF NOT EXISTS users ' + 
      '(discord_id TEXT PRIMARY KEY, sekai_id TEXT, private INTEGER DEFAULT 1, ' + 
      'quiz_correct INTEGER DEFAULT 0, quiz_question INTEGER DEFAULT 0)').run()

    // Initialize the tracking database instance
    this.db.prepare('CREATE TABLE IF NOT EXISTS tracking ' + 
      '(channel_id TEXT PRIMARY KEY, guild_id TEXT, tracking_type INTEGER)').run()
  }
```

I would recommend storing all of this information in it's own .db file.

Thanks Ai0!